### PR TITLE
POI Tree Rotations Editorial

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

NOTE: I changed the OJ to oj.uz, because the other one is more annoying to use IMO.

### CODE REVIEW

**Comment by TheGamingMousse** [comment]
https://oj.uz/submission/1085231 AC (same code, just less spacing)

File: solutions/platinum/poi-11-TreeRotations.mdx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using ll = long long;
+
+#include <ext/pb_ds/assoc_container.hpp>
+using namespace __gnu_pbds;
```
Comment (SansPapyrus683) [inline solutions/platinum/poi-11-TreeRotations.mdx:36]: how many things you need are in this namespace
seems like you just take `Tree` from it

File: solutions/platinum/poi-11-TreeRotations.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			for (int v : sets[idx_l]) { sets[idx_r].insert(v); }
+			sets[idx].swap(sets[idx_r]);
+			sets.erase(idx_l);
+			sets.erase(idx_r);
+
+			inversions[idx] =
+			    inversions[idx_l] + inversions[idx_r] + min(way_1, way_2);
+		} else {
+			sets[idx].insert(val);
+		}
```
Comment (SansPapyrus683) [inline solutions/platinum/poi-11-TreeRotations.mdx:79]: can you make this explicit with an else if

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/platinum/poi-11-TreeRotations.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/platinum/poi-11-TreeRotations.mdx:36]: `null_type`, `rb_tree_tag`, `tree_order_statistics_node_update`, and `tree` are all from it

I'm ngl I just copied and pasted the ordered statistics declaration stuff thing from the module lmao

Comment (SansPapyrus683) [reply solutions/platinum/poi-11-TreeRotations.mdx:36]: yeah understandable

**Review by SansPapyrus683 — APPROVED** [review]
definitely not me not even looking at the explanation

**Comment by TheGamingMousse** [comment]
> definitely not me not even looking at the explanation

LOL

**Review by ryanchou-dev — CHANGES_REQUESTED** [review]
ur code looks better and better

**Comment by TheGamingMousse** [comment]
> ur code looks better and better

WWWW

File: solutions/platinum/poi-11-TreeRotations.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
+---
+id: poi-11-TreeRotations
+source: POI
+title: 2011 - Tree Rotations
+author: Justin Ji
+---
+
```
[the remark this answers is not in the document]
Comment (TheGamingMousse) [reply solutions/platinum/poi-11-TreeRotations.mdx:7]: holyyyy how do i keep on forgetting this LMFAO

[the remark this answers is not in the document]
Comment (TheGamingMousse) [reply solutions/platinum/poi-11-TreeRotations.mdx:7]: ok added it, altho it's in polish

File: solutions/platinum/poi-11-TreeRotations.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+together is the sum of the inversions within the two sets plus the
+number of new inversions after joining the two sets, with each new inversion
+having one item in the left and right sets. Thus, to count the
+best way to merge the two sets, we need to find the inversions if we either swap
+or don't swap the two sets.
+
+The implementation below uses small-to-large merging and ordered sets to
+efficiently count the least inversions we can have when combining the sets from
+our left and right subtrees. We iterate on the smaller set, and use the ordered
+set `order_of_key` method to count the number of elements $< x$ and $> x$, for
```
[the remark this answers is not in the document]
Comment (TheGamingMousse) [reply solutions/platinum/poi-11-TreeRotations.mdx:22]: ordered set / indexed set / ordered statistics tree are all pretty commonly used I think

**Comment by SansPapyrus683** [comment]
> ur code looks better and better

implying it wasn't already goated

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**